### ISSUE TITLE
Better find "the frontier"

### ISSUE THREAD

**Issue by alecramsay** [rdatools/rdabase#44]
Some ideas to better approximate DRA's human-drawn notable maps and find "the frontier."

We currently "push" the points near the frontier of an ensemble, requiring that each mutation improve the (continuous) ratings on one or both dimensions. 

Alternatively, we could require that each mutation improve only one dimension while the other stays w/in some threshold (e.g., 1%) of the original value. This is the "pin" functionality that I've implemented. So, this option would pin one dimension and optimize the other. We could push each point in both directions some number of times. By uncoupling the dimensions, we might get higher maxima.

We could also increase the threshold (elasticity) in the pinned dimension, the idea being giving the solver more flexibility to find solutions. For example, instead of a 1% threshold, we could use 5% or even 5 ratings units.

**Comment by alecramsay** [comment]
Or we could explore Moon's idea for "pushing" the frontier [using ReCom](https://mggg.org/posts/gerrysuite )

**Comment by alecramsay** [comment]
This seems to have worked like gang busters. Check out [the new plots for NC](https://rdatools.github.io/tradeoffs/states/NC/).

The only place where the method falls short of the human-drawn Notable Maps from DRA is on compactness, and that map splits a bunch of precincts to smooth out district boundaries.

I should run another state or two before declare victory, but I think this more than illustrates our contention / hypothesis that there are clear trade offs between redistricting criteria.

**Comment by alecramsay** [comment]
A few notes about what I did:

- I first got SingleMetricOptimizer working for Polsby-Popper, which was one of their marquee examples -- This took some doing, as the examples were unclear and incomplete and used a different method than another example that I'd previously cloned to figure out how to prep the data that GerryChain needs. Once I got a complete example notebook from Moon's key dev, Peter, I was off to the races.
- I first tried using the root map -- the seed to the unbiased ensembles -- as the starting point for optimizing. That didn't seem to work too well.
- Then I tried following the example of starting with random initial assignments of precinct to districts. That also didn't do too well.
- Then I realized that essentially what we wanted to do was connect each pair of Notable Maps with a frontier of points, so I optimized each of Notable Maps for each of the 5 ratings dimensions.
- After I'd gotten compactness (using Polsby-Popper as a proxy) working, adding metrics for the other dimensions was pretty easy--some jiggering here and there, but it was pretty easy to figure out.
- I modified the workflow to take in these optimized ensembles, find the "pushed" frontiers and do all the downstream plotting and such.

So, the obvious _assist_ that I gave the process was seeding the optimization runs with the Notable Maps. (I had copied these and removed split precincts, so could be used by my runs which have VTD granulariy. They're close but not quite the Notable Maps.)
